library/ssl_cli.c

```diff
@@ -4310,6 +4338,29 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
            break;
 
        case MBEDTLS_SSL_CLIENT_FINISHED:
+
+#if defined(MBEDTLS_DELAYED_SERVER_CERT_VERIFICATION)
+#if defined(MBEDTLS_SSL_SRV_C) && defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
+           authmode = ssl->handshake->sni_authmode != MBEDTLS_SSL_VERIFY_UNSET
+                       ? ssl->handshake->sni_authmode
+                       : mbedtls_ssl_conf_get_authmode( ssl->conf );
+#else
+           authmode = mbedtls_ssl_conf_get_authmode( ssl->conf );
+#endif
+/*           authmode = ssl->handshake->sni_authmode != MBEDTLS_SSL_VERIFY_UNSET
+                       ? ssl->handshake->sni_authmode
+                       : ssl->conf->authmode;
+*/
+           chain = ssl->session_negotiate->peer_cert;
+
+           MBEDTLS_SSL_DEBUG_MSG( 3, ( "execute delayed server certificate verification" ) );
+
+           ret = mbedtls_ssl_parse_delayed_certificate_verify( ssl, authmode,
```

Comment:
Added.

Reply:
I mean before line 4262, because if it is initialized only in variable definition, it's value will be zero after lines   4242 and 4249

Reply:
There are states that do not update the ret variable, and they would fail if I added it there. I can change ret just before `delayed_certificate_verify`, but I'm not sure if this makes sense, as we would be running verification after it anyway.